subsys/shell/shell_uart.c

```diff
@@ -218,12 +246,19 @@ static int read(const struct shell_transport *transport,
 
 	return 0;
 }
+static void update(const struct shell_transport *transport)
+{
+	struct shell_uart *sh_uart = (struct shell_uart *)transport->ctx;
+
+	smp_shell_process(&sh_uart->ctrl_blk->smp);
+}
 
 const struct shell_transport_api shell_uart_transport_api = {
 	.init = init,
 	.uninit = uninit,
 	.enable = enable,
 	.write = write,
+	.update = update,
```

Comment:
Initialization is not in the order of structure definition. This will cause a warning for C++ compiler.